# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/mp4/MP4Demuxer.cpp
# Commit: 93cf96e033d0
# Full Hash: 93cf96e033d0d7c13b66a279bb4cb1f7138dd473
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-05-31 21:43:54
# Description:
#   Bug 1833896 - Consistently use uint32_t to store timescale until it reaches the TimeUnit constructor, which uses int64_t. r=media-playback-reviewers,karlt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D179363
# ==============================================================================

diff -r 5b7624f7163b -r 93cf96e033d0 dom/media/mp4/MP4Demuxer.cpp
--- a/dom/media/mp4/MP4Demuxer.cpp	Wed May 31 09:35:17 2023 +0000
+++ b/dom/media/mp4/MP4Demuxer.cpp	Wed May 31 09:36:03 2023 +0000
@@ -44,7 +44,7 @@
                         public DecoderDoctorLifeLogger<MP4TrackDemuxer> {
  public:
   MP4TrackDemuxer(MediaResource* aResource, UniquePtr<TrackInfo>&& aInfo,
-                  const IndiceWrapper& aIndices, int32_t aTimeScale);
+                  const IndiceWrapper& aIndices, uint32_t aTimeScale);
 
   UniquePtr<TrackInfo> GetInfo() const override;
 
@@ -313,7 +313,7 @@
 MP4TrackDemuxer::MP4TrackDemuxer(MediaResource* aResource,
                                  UniquePtr<TrackInfo>&& aInfo,
                                  const IndiceWrapper& aIndices,
-                                 int32_t aTimeScale)
+                                 uint32_t aTimeScale)
     : mResource(aResource),
       mStream(new ResourceStream(aResource)),
       mInfo(std::move(aInfo)),